src/main/java/com/ericsson/ei/frontend/utils/BackEndInstancesUtils.java

```diff
@@ -16,134 +16,220 @@
 */
 package com.ericsson.ei.frontend.utils;
 
-import com.ericsson.ei.frontend.model.BackEndInformation;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.gson.*;
-import com.google.gson.reflect.TypeToken;
-import lombok.Getter;
-import lombok.Setter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
-import javax.annotation.PostConstruct;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.List;
+import com.ericsson.ei.frontend.model.BackEndInformation;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import lombok.Getter;
+import lombok.Setter;
 
 @Getter
 @Setter
 @Component
 public class BackEndInstancesUtils {
 
     private static final Logger LOG = LoggerFactory.getLogger(BackEndInstancesUtils.class);
-    private static final String PATH_TO_WRITE = "src/main/resources/EIBackendInstancesInformation.json";
     private static final String NAME = "name";
     private static final String HOST = "host";
     private static final String PORT = "port";
     private static final String PATH = "path";
     private static final String HTTPS = "https";
-    private static final String ACTIVE = "active";
 
-    @Value("${ei.backendServerHost:#{null}}")
-    private String host;
+    @Value("${ei.backendServerName:#{null}}")
+    private String defaultBackEndInstanceName;
 
-    @Value("${ei.backendServerPort:#{null}}")
-    private String port;
+    @Autowired
+    private BackEndInformation defaultBackendInformation;
 
-    @Value("${ei.backendContextPath:#{null}}")
-    private String path;
+    @Autowired
+    private BackEndInstanceFileUtils backEndInstanceFileUtils;
+
+    private List<BackEndInformation> backEndInformationList = new ArrayList<>();
+
+    /**
+     * Function to check weather an instance host, port, path and https is unique.
+     *
+     * @param instance
+     *      JsonObject
+     * @return
+     *      boolean
+     */
+    public boolean checkIfInstanceAlreadyExist(JsonObject instance) {
+        parseBackEndInstances();
```

Comment:
How will this work? Will each sessions have an own Instance of this class? Or how do you handle different EI backend instance addresses per sessions?

Reply:
All users have access to see available instances added by any user, they are stored in a common file that is accessed every time instances needs to be read. An update from one user is seen by another. Only the instance a user uses is set to be unique for a specific session.

Reply:
If a user goes to another page(Not EI-Frontend, e.g. google.com) and then to EI-Frontend again, will default EI-Backend be used or do it remember last used EI-Backend for each user/old session?

Reply:
As I understand it the session is bound to the browser. I will test this again to be sure.

Reply:
After a test the session is bound to browser, unless you open a "private window" in Chrome it will remember the setting.